### ISSUE TITLE
Possible ITE: Resolve attestation references during verification

### ISSUE THREAD

**Issue by adityasaky** [in-toto/ITE#53]
Some ITE-6 attestations like [SCAI](https://github.com/in-toto/attestation/blob/main/spec/predicates/scai.md) and [VSA](https://github.com/in-toto/attestation/blob/main/spec/predicates/vsa.md) can reference other attestations. During verification, it may be necessary to resolve these references and apply rules against the target attestations.

Existing work: @marcelamelara has prototyped some support for this attestation resolving behaviour for SCAI. As a next step, we should generalize this for resolving references in all predicates and formally specify how it works. This may require a new ITE, alongside ITEs 10 and 11 which update the layout schema to support attestations.

This was first raised by @MarkLodato during a review of ITE-11: https://github.com/in-toto/ITE/pull/50#discussion_r1344482464

**Comment by trishankatdatadog** [comment]
Interesting problem, but this may be going down a tricky road unless we have things like [PURLs](https://en.wikipedia.org/wiki/Persistent_uniform_resource_locator). I think it's up to clients to pass in a complete bundle to verifiers, but would be good to consider other options.

**Comment by adityasaky** [comment]
Right, one thing we'd have to do is work out the scope for this. Note that in the simplest case, this could be scoped precisely to the bundle passed to a verifier. Even then, if the verification encounters a SCAI predicate, the layout may call for resolving the evidence for some attribute to the full attestation _in the same bundle_ to apply other rules.
